### ISSUE TITLE
SST random failure in evpath

### ISSUE THREAD

**Issue by pnorbert** [ornladios/ADIOS2#413]
Heat transfer example regularly fails with anything larger than 4 to 1 example on my VM.
```bash
mpirun -n 6 ./bin/heatTransfer_write_adios2 /home/adios/work/ADIOS2/examples/heatTransfer/heat_sst.xml insitu.bp 3 2  10 10 10 1
```
and
```bash
mpirun -n 1 ./bin/heatTransfer_read /home/adios/work/ADIOS2/examples/heatTransfer/heat_sst.xml insitu.bp sst.bp 1 1
```

Writer fails at some steps randomly. 5 or 6 writers exhibit this problem almost always, 4 writers sometimes.

```
...
Variable T
Shape ID GlobalArray : putshape  10 10
varshape  30 20
offsets  20 0
[adiosVM:29113] *** Process received signal ***
[adiosVM:29113] Signal: Segmentation fault (11)
[adiosVM:29113] Signal code: Address not mapped (1)
[adiosVM:29113] Failing at address: (nil)
[adiosVM:29113] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7fb92c51a390]
[adiosVM:29113] [ 1] /opt/adios2/lib/libsst.so(dequeueTimestep+0x4a)[0x7fb92b15755a]
[adiosVM:29113] [ 2] /opt/adios2/lib/libsst.so(CP_ReleaseTimestepHandler+0xad)[0x7fb92b15770c]
[adiosVM:29113] [ 3] /opt/chaos/lib/libevpath.so.4(+0x20950)[0x7fb92981b950]
[adiosVM:29113] [ 4] /opt/chaos/lib/libevpath.so.4(CMDataAvailable+0x26a)[0x7fb92981c91a]
[adiosVM:29113] [ 5] /opt/chaos/lib/libcmenet.so(+0x1dd4)[0x7fb911545dd4]
[adiosVM:29113] [ 6] /opt/chaos/lib/libcmenet.so(+0x2119)[0x7fb911546119]
[adiosVM:29113] [ 7] /opt/chaos/lib/libcmselect.so(+0x1758)[0x7fb912156758]
[adiosVM:29113] [ 8] /opt/chaos/lib/libevpath.so.4(CMcontrol_list_wait+0xa4)[0x7fb929812284]
[adiosVM:29113] [ 9] /opt/chaos/lib/libevpath.so.4(+0x17c2c)[0x7fb929812c2c]
[adiosVM:29113] [10] /opt/chaos/lib/libevpath.so.4(+0x17f89)[0x7fb929812f89]
[adiosVM:29113] [11] /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba)[0x7fb92c5106ba]
[adiosVM:29113] [12] /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7fb92b6cf41d]
```

**Comment by eisenhauer** [comment]
Actually a problem in SST I think.  I believe that this is symptomatic of the bug fixed by the patch I submitted to your WIP on common staging testing, here:
https://github.com/ornladios/ADIOS2/pull/399/commits/163111e369a3ab22fbff74a9eec50aaa57ad902c
I may have done the wrong github thing somehow there.  Has it created a new PR that has just this fix along with the commits in your PR?   We could go ahead and apply this fix separately...  You want to cherry-pick it and see if it fixes this particular issue?

**Comment by pnorbert** [comment]
Fixed by PR #424
